src/components/SignUp/CareerInfo.js

```diff
@@ -74,6 +75,20 @@ class CareerInfo extends Component {
       Actions.refresh({ rightTitle: 'SAVE', onRight: this.onNextBtnPressed.bind(this) });
   }
 
+  componentWillReceiveProps(props) {
+    if(props.fromEdit && this.state.needRefresh) {
```

Comment:
fromEdit is missing in prop validation. well, regardless of the validation, what if props does not have fromEdit property? are you sure this always get fromEdit prop?